src/models/core/AnswersConfig.ts

```diff
@@ -42,44 +40,12 @@ export interface BaseAnswersConfig {
   }
 }
 
-
 /**
  * Configuration options for {@link AnswersCore}, which includes the
- * options from {@link BaseAnswersConfig}, but requires apiKey.
+ * options from {@link AnswersConfig}, but requires token instead of apiKey.
  * @public
  */
-export interface AnswersConfigWithApiKey extends BaseAnswersConfig {
-  /**
-   * {@inheritDoc BaseAnswersConfig.apiKey}
-   */
-  apiKey: string,
-  /**
-   * token should NOT be provided along with apiKey
-   */
-  token?: never
-}
-
-/**
- * Configuration options for {@link AnswersCore}, which includes the
- * options from {@link BaseAnswersConfig}, but requires token.
- * @public
- */
-export interface AnswersConfigWithToken extends BaseAnswersConfig {
-  /**
-   * {@inheritDoc BaseAnswersConfig.token}
-   */
-  token: string,
-  /**
-   * apiKey should NOT be provided along with token
-   */
-  apiKey?: never
+export interface AnswersConfigWithToken extends Omit<AnswersConfig,'apiKey'> {
```

Comment:
nit: space after comma

Reply:
I'm not a big fan of having an `AnswersConfig` and an `AnswersConfigWithToken`. As a consumer, I'd expect there to be a single configuration object. It feels a bit odd to make a whole new configuration object that's exactly the same, except with one property changed.

Reply:
maybe could do some type composition like the below?
```ts
// unsure about all naming
interface AnswersApiKey = { apiKey: string }
interface AnswersToken = { token: string }
export type AnswersConfighWithAuth = (AnswersConfig & AnswersApiKey) | (AnswersConfig & AnswersToken)
```

Reply:
there's two other ways we could do:
1) have both in AnswersConfig as optional. But this loses the typescript check and we would have to manually check and throw error to user in the different services if both are defined or undefined.
```
 apiKey?: string
 token?: string
```
2) have something similar to visitor. Where both fields are required, and user specified what to use. But this wouldn't be backward compatible.
```
authMethod: 'API_KEY' or 'TOKEN'
authId: string
```

Reply:
@oshi97, that would uses type instead of interface, which is the breaking change that connor was talking about. I'm not sure if we would want that?

Reply:
I'm okay with a breaking change. This isn't the best justification, but I believe that few people are using this library besides us, and imagine that 100% of them are within Yext. I agree with Tom that `AnswersConfigWithToken` is a little strange.
In my mind the goals here are
1. refactor `AnswersConfig` in a way that won't need to be broken later on (or at least broken less)
2. still have a nice interface for people to use. i.e., if we didn't have to worry about breaking anything, what would we want things to look like? 

To that end I think what we had before does that a little better, though I imagine there might be some improvements we can still make? I tried looking at some other libraries like the [firebase js sdk](https://github.com/firebase/firebase-js-sdk/blob/master/packages/app/src/public-types.ts) and [algolia's js client](https://github.com/algolia/algoliasearch-client-javascript/tree/master/packages/client-common/src/types) but couldn't really find anything obvious we were missing. Not sure if @cea2aj has any ideas here?

Reply:
With the previous approach, we had 4 documentation pages for Answers configs (AnswersConfig, AnswersConfigWithToken, AnswersConfigWithApiKey, and BaseAnswersConfig). I figured this would be simpler with just 2 documentaiton pagees. Plus this allows us to remove the `never` keyword from the interfaces. The previous appraoch would enforce that only one type of auth is used for consumers of Core, however it would not do so within the library which is a bit strange. Here we have much stronger type checking at build time. We are also avoiding all breaking changes with this approach.

If we aren't concerned about breaking changes, then perhaps we could use the previous approach but with some modifications as you suggested. Maybe instead of using the 'never' keyword, we could use the `Omit` utility type to omit the auth method which isn't being used. That might give us the stronger typing as well

Reply:
One idea I have for making the previous approach less of a breaking change would be to keep the `AnswersConfig` as an interface (but with apiKey and token marked as optional), and then we introduce a new type such as `SpecificAnswersConfig` which is the new `type` passed to `provideCore`